internal/backends/python/search_pypi.go

```diff
@@ -0,0 +1,97 @@
+package python
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/replit/upm/internal/api"
+	"golang.org/x/net/html"
+)
+
+var WhiteSpaceChars = " \n\t"
+
+// Port of https://github.com/asadmoosvi/pypi-search/blob/main/pypi_search/search.py
+func SearchPypi(query string) ([]api.PkgInfo, error) {
+	endpoint := fmt.Sprintf("https://pypi.org/search/?q=%s", url.QueryEscape(query))
+	req, err := http.NewRequest("GET", endpoint, nil)
```

Comment:
Can you use your http client definitions in api/http.go instead? They will use our user-agent.